src/core/namespace.c

```diff
@@ -1812,6 +1812,37 @@ static int create_symlinks_from_tuples(const char *root, char **strv_symlinks) {
         return 0;
 }
 
+static void mount_entry_path_debug_string(const char *root, MountEntry *m, char **error_path) {
+        assert(m);
+
+        /* Create a string suitable for debugging logs, stripping for example the local working directory.
+         * For example, with a BindPaths=/var/bar that does not exist on the host:
+         *
+         * Before:
+         *  foo.service: Failed to set up mount namespacing: /run/systemd/unit-root/var/bar: No such file or directory
+         * After:
+         *  foo.service: Failed to set up mount namespacing: /var/bar: No such file or directory
+         *
+         * Note that this is an error path, so no OOM check is done on purpose. */
+
+        if (!error_path || !mount_entry_path(m))
+                return;
```

Comment:
actually, if you have a helper call here now that initializes things in more detail, maybe we should start initializing error_path to NULL if we have no data. i.e. this returns an error value of sorts, and it would make sense to always initialize it...